core/src/test/scala/quasar/destination/s3/S3DestinationSpec.scala

```diff
@@ -97,10 +97,10 @@ object S3DestinationSpec extends EffectfulQSpec[IO] {
 }
 
 final class MockUpload(status: Ref[IO, Map[ObjectKey, String]]) extends Upload[IO] {
-  def upload(bytes: Stream[IO, Byte], bucket: Bucket, key: ObjectKey): IO[Unit] =
+  def upload(bytes: Stream[IO, Byte], bucket: Bucket, key: ObjectKey): Stream[IO, Unit] =
     for {
-      data <- bytes.through(text.utf8Decode).compile.string
-      _ <- status.update(_ + (key -> data))
+      data <- bytes.through(text.utf8Decode)
```

Comment:
If `bytes` contains more than chunk, this `flatMap` will run multiple times.

Reply:
I have fixed it now. It should be okay if there's more than one chunk. It will concat the string for each new chunk.